core/src/test/java/org/apache/iceberg/catalog/CatalogTransactionTests.java

```diff
@@ -0,0 +1,736 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.catalog;
```

Comment:
Many of these tests are good validations that the transaction is isolated. You might want to separate these into more specific suites, like tests of isolation for uncommitted transactions. This is going to get pretty large otherwise.

What's missing are tests of real-world cases that will help us look at the behavior of transactions. For that, I think it would be helpful to refactor and introduce methods with high-level names, like `createTestTable(A_IDENT)` or `removeColumn(t.load(A_IDENT), "data")`.